src/main/java/dynamok/sink/DynamoDbSinkTask.java

```diff
@@ -70,7 +71,26 @@ String topAttributeName(ConnectorConfig config) {
     @Override
     public void start(Map<String, String> props) {
         config = new ConnectorConfig(props);
-        client = new AmazonDynamoDBClient();
+
+            // TODO : Verify functionality of 
+            //      SystemPropertiesCredentialsProvider
+        if (config.accessKeyId == null  ||  config.secretKeyId == null) {
```

Comment:
This branch is unreachable since the default values for `access.key.id` and `secret.key.id` are `""` -- they can only ever be `null` if that is the default value and the user does not override.

In any case, I don't think it's worthwhile to have this branch. In the second case where the access key or secret key is empty, that will end up using the `DefaultAWSCredentialsProviderChain` that subsumes both `EnvironmentVariableCredentialsProvider` and `SystemPropertiesCredentialsProvider` (and a few other fallbacks: http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#id6)